### PR TITLE
Fix race condition in the gateway doc deletion

### PR DESCRIPTION
**Pull request by bennerv**
### Which issue this PR addresses:

Fixes a race condition in the gateway enablement.  If we delete the PE, there's a chance another cluster could spin up a PE and get the linkID reused.  

### What this PR does / why we need it:

* Fixes a gateway race condition
* Disables egress on new cluster create
* Removes the `gatewayDomains` from `cluster.aro.openshift.io` resource on `AdminUpdate` essentially removing the dnsmasq config that makes egress work.  

### Test plan for issue:

* Do it

### Is there any documentation that needs to be updated for this PR?
* No

### CODE REVIEW

**Review by jewzaam — APPROVED** [review]

File: pkg/cluster/ipaddresses.go [hunk 1]
```diff
@@ -255,10 +255,25 @@ func (m *manager) ensureGatewayCreate(ctx context.Context) error {
...
 	})
-	if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusConflict) /* already exists */ {
+
+	recordExists := err != nil && cosmosdb.IsErrorStatusCode(err, http.StatusConflict)
+	if err != nil && !recordExists /* already exists */ {
 		return err
 	}
 
+	// ensure the record is this clusters if it exists
+	if recordExists {
```
Comment (karanmagdani) [inline pkg/cluster/ipaddresses.go:265]: @bennerv When would this condition happen? If this is during cluster create, don't we just need to check that the record does not exist, and if it doesn't we create a new record

Comment (25region) [reply pkg/cluster/ipaddresses.go:265]: @karanmagdani The gw record is created before this step and we need to ensure that it is for this specific cluster and it is not a stale/legacy record that is leftover.

https://github.com/Azure/ARO-RP/pull/1827/files/7b70040dd4e23084e4659ec6514ee083e8021842#diff-7369cd43feeb39bc03ab9c03efa6c5f85621d8554b94381e7d0a88398aca79a4R250

Comment (karanmagdani) [reply pkg/cluster/ipaddresses.go:265]: Ah got it, makes sense! thank you @25region

**Review by 25region — APPROVED** [review]

File: pkg/operator/deploy/deploy.go [hunk 2]
```diff
@@ -184,18 +184,10 @@ func (o *operator) resources() ([]kruntime.Object, error) {
 		},
 	}
 
-	if o.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP != "" {
-		cluster.Spec.GatewayDomains = append(o.env.GatewayDomains(), o.oc.Properties.ImageRegistryStorageAccountName+".blob."+o.env.Environment().StorageEndpointSuffix)
-	} else {
-		cluster.Spec.InternetChecker = arov1alpha1.InternetCheckerSpec{
-			URLs: []string{
-				fmt.Sprintf("https://%s/", o.env.ACRDomain()),
```
Comment (rogbas) [inline pkg/operator/deploy/deploy.go:192]: From our sync I understood we would still need to populate `spec.InternetChecker`, is that not the case?

Comment (bryanro92) [reply pkg/operator/deploy/deploy.go:192]: It's still populated from https://github.com/Azure/ARO-RP/blob/7b70040dd4e23084e4659ec6514ee083e8021842/pkg/operator/deploy/deploy.go#L160-L166

Comment (bryanro92) [reply pkg/operator/deploy/deploy.go:192]: This else statement was apparently redundant and not needed

Comment (rogbas) [reply pkg/operator/deploy/deploy.go:192]: thanks for clarifying Ross

**Review by rogbas — APPROVED** [review]

**Comment by bennerv** [comment]
Merging this as it's now released to all regions and rolled out.

**Pull request merged**